{"trial_id": "0020", "hyperparameters": {"space": [{"class_name": "Int", "config": {"name": "n_hidden", "default": 10, "conditions": [], "min_value": 1, "max_value": 50, "step": 1, "sampling": "linear"}}, {"class_name": "Int", "config": {"name": "n_neurons", "default": null, "conditions": [], "min_value": 10, "max_value": 100, "step": 1, "sampling": "linear"}}, {"class_name": "Float", "config": {"name": "learning_rate", "default": 0.0001, "conditions": [], "min_value": 0.0001, "max_value": 0.1, "step": null, "sampling": "log"}}, {"class_name": "Choice", "config": {"name": "optimizer", "default": "sgd", "conditions": [], "values": ["sgd", "adam"], "ordered": false}}, {"class_name": "Choice", "config": {"name": "hidden_activation_func", "default": "relu", "conditions": [], "values": ["relu", "leaky_relu", "elu", "gelu", "swish", "mish"], "ordered": false}}, {"class_name": "Int", "config": {"name": "n_clusters", "default": null, "conditions": [], "min_value": 2, "max_value": 10, "step": 1, "sampling": "linear"}}, {"class_name": "Int", "config": {"name": "batch_size", "default": null, "conditions": [], "min_value": 10, "max_value": 2000, "step": 1, "sampling": "linear"}}, {"class_name": "Boolean", "config": {"name": "normalize", "default": false, "conditions": []}}], "values": {"n_hidden": 19, "n_neurons": 64, "learning_rate": 0.0804925654160973, "optimizer": "adam", "hidden_activation_func": "mish", "n_clusters": 2, "batch_size": 1281, "normalize": true, "tuner/epochs": 4, "tuner/initial_epoch": 0, "tuner/bracket": 1, "tuner/round": 0}}, "metrics": {"metrics": {}}, "score": null, "best_step": 0, "status": "FAILED", "message": "Traceback (most recent call last):\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras_tuner/src/engine/base_tuner.py\", line 274, in _try_run_and_update_trial\n    self._run_and_update_trial(trial, *fit_args, **fit_kwargs)\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras_tuner/src/engine/base_tuner.py\", line 239, in _run_and_update_trial\n    results = self.run_trial(trial, *fit_args, **fit_kwargs)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras_tuner/src/tuners/hyperband.py\", line 427, in run_trial\n    return super().run_trial(trial, *fit_args, **fit_kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras_tuner/src/engine/tuner.py\", line 314, in run_trial\n    obj_value = self._build_and_fit_model(trial, *args, **copied_kwargs)\n                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras_tuner/src/engine/tuner.py\", line 233, in _build_and_fit_model\n    results = self.hypermodel.fit(hp, model, *args, **kwargs)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/johannes/Essentials_Masters/MSc_Bioinformatics/MSc_Research_Project/ActualProjectDevelopment/MScProjectMachineLearning/MachineLearningClusteringMScProject/scripts/clustering/general_deep_learning.py\", line 66, in fit\n    return neural_model_unsup.fit(X, y, batch_size=batch_size, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/utils/traceback_utils.py\", line 122, in error_handler\n    raise e.with_traceback(filtered_tb) from None\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/tensorflow/python/eager/execute.py\", line 59, in quick_execute\n    except TypeError as e:\ntensorflow.python.framework.errors_impl.InvalidArgumentError: Graph execution error:\n\nDetected at node compile_loss/sparse_categorical_crossentropy/SparseSoftmaxCrossEntropyWithLogits/SparseSoftmaxCrossEntropyWithLogits defined at (most recent call last):\n  File \"/home/johannes/Essentials_Masters/MSc_Bioinformatics/MSc_Research_Project/ActualProjectDevelopment/MScProjectMachineLearning/MachineLearningClusteringMScProject/scripts/clustering/deep_learning_clustering_hits.py\", line 185, in <module>\n\n  File \"/home/johannes/Essentials_Masters/MSc_Bioinformatics/MSc_Research_Project/ActualProjectDevelopment/MScProjectMachineLearning/MachineLearningClusteringMScProject/scripts/clustering/deep_learning_clustering_hits.py\", line 167, in main\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras_tuner/src/engine/base_tuner.py\", line 234, in search\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras_tuner/src/engine/base_tuner.py\", line 274, in _try_run_and_update_trial\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras_tuner/src/engine/base_tuner.py\", line 239, in _run_and_update_trial\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras_tuner/src/tuners/hyperband.py\", line 427, in run_trial\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras_tuner/src/engine/tuner.py\", line 314, in run_trial\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras_tuner/src/engine/tuner.py\", line 233, in _build_and_fit_model\n\n  File \"/home/johannes/Essentials_Masters/MSc_Bioinformatics/MSc_Research_Project/ActualProjectDevelopment/MScProjectMachineLearning/MachineLearningClusteringMScProject/scripts/clustering/general_deep_learning.py\", line 66, in fit\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/utils/traceback_utils.py\", line 117, in error_handler\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/backend/tensorflow/trainer.py\", line 320, in fit\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/backend/tensorflow/trainer.py\", line 121, in one_step_on_iterator\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/backend/tensorflow/trainer.py\", line 108, in one_step_on_data\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/backend/tensorflow/trainer.py\", line 54, in train_step\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/trainers/trainer.py\", line 359, in _compute_loss\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/trainers/trainer.py\", line 327, in compute_loss\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/trainers/compile_utils.py\", line 611, in __call__\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/trainers/compile_utils.py\", line 652, in call\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/losses/loss.py\", line 60, in __call__\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/losses/losses.py\", line 27, in call\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/losses/losses.py\", line 1870, in sparse_categorical_crossentropy\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/ops/nn.py\", line 1559, in sparse_categorical_crossentropy\n\n  File \"/home/johannes/Practice/MachineLearning/HandsOnMachineLearning/lib/python3.12/site-packages/keras/src/backend/tensorflow/nn.py\", line 671, in sparse_categorical_crossentropy\n\nReceived a label value of 3 which is outside the valid range of [0, 3).  Label values: 0 1 1 0 0 0 1 0 1 0 1 1 3 0 1 1 1 1 0 0 1 1 0 0 0 1 1 0 1 1 3 0 3 0 0 0 1 1 0 0 0 0 1 0 1 0 3 3 1 0 0 3 1 1 1 1 1 0 2 0 1 2 0 1 3 1 1 0 0 1 0 0 0 0 1 1 1 1 1 3 0 2 3 2 3 0 1 1 3 1 3 1 0 3 1 1 3 0 0 0 3 1 0 1 2 0 1 1 1 2 1 2 1 1 1 0 0 3 0 0 1 0 1 1 1 1 1 3 0 0 3 1 1 0 0 0 0 0 1 1 2 0 1 3 0 3 1 0 1 3 1 0 1 0 0 0 1 1 0 1 1 0 0 1 1 0 3 0 0 2 1 0 1 1 2 1 1 0 0 2 2 0 0 1 1 0 0 2 0 0 0 1 0 3 1 0 0 1 0 0 0 0 1 0 1 0 3 2 0 0 0 0 0 0 0 1 2 3 0 0 1 0 0 0 1 0 0 3 0 1 3 1 0 1 1 3 0 0 0 0 1 0 0 1 1 2 0 0 2 1 1 1 0 0 1 1 0 3 3 1 1 0 0 0 1 0 1 1 1 0 0 3 2 1 1 0 3 1 0 1 0 2 1 1 1 0 1 1 0 1 1 0 1 1 1 0 1 1 0 1 1 1 0 0 3 1 1 1 1 0 0 0 1 1 0 0 0 0 0 2 1 3 1 1 1 0 1 1 0 0 0 1 1 1 1 1 2 2 0 1 0 2 2 1 1 0 0 1 0 0 1 2 1 0 1 1 1 0 2 1 0 1 1 1 1 1 3 0 3 1 1 0 0 0 1 1 0 0 2 0 0 0 0 3 3 1 1 0 1 0 0 1 1 1 0 1 0 1 3 0 1 3 0 1 0 0 0 2 1 1 1 0 1 1 3 1 1 2 2 1 0 2 1 0 0 0 2 2 1 1 3 0 0 3 1 0 0 0 2 1 1 1 0 0 3 0 1 2 3 0 0 1 0 0 1 1 1 0 0 1 0 0 1 1 1 1 1 1 0 0 1 1 0 1 1 1 3 1 0 0 1 0 0 1 3 1 1 1 3 1 0 1 1 0 1 1 0 1 0 0 1 1 1 0 1 0 1 0 1 0 0 0 3 0 1 0 0 1 1 1 1 1 1 0 0 0 1 0 1 0 2 0 0 1 1 1 1 1 0 1 1 1 0 3 1 0 0 0 0 0 0 0 0 1 0 1 0 3 1 1 0 1 1 0 1 3 1 1 1 1 1 0 0 1 1 0 2 3 0 0 1 0 0 0 2 1 1 1 0 0 1 1 0 1 1 1 0 1 1 3 0 1 1 2 3 1 1 2 1 1 3 1 0 3 0 3 0 1 0 0 2 1 1 1 3 1 0 0 0 0 3 1 1 1 0 0 1 0 0 0 1 3 0 0 2 1 3 1 1 1 2 0 1 2 0 0 1 1 0 1 3 1 0 1 3 1 1 0 1 1 1 2 0 1 1 0 1 0 2 1 1 0 0 1 1 0 1 1 1 0 2 3 0 1 0 1 0 1 1 3 1 1 1 2 1 1 1 0 1 1 0 0 1 0 1 0 0 0 1 1 0 3 2 2 1 1 1 0 0 0 0 3 0 0 0 1 0 0 2 2 1 1 1 0 1 0 3 1 1 0 0 0 0 0 2 1 0 2 1 1 1 3 0 0 0 3 0 1 2 1 0 1 0 1 1 0 0 1 0 0 1 0 1 2 0 0 0 3 1 0 0 2 3 0 0 3 1 0 3 0 0 1 2 0 1 1 1 0 1 1 3 0 0 2 1 1 3 1 1 3 2 3 0 0 2 3 1 3 1 1 1 0 0 0 1 1 0 1 0 1 0 1 0 1 0 3 1 1 1 0 1 0 1 1 1 1 1 0 1 3 1 0 0 1 1 0 0 2 1 1 1 0 1 1 1 0 2 0 1 0 1 1 2 0 0 0 1 3 2 2 3 1 1 3 1 1 1 1 1 1 3 3 2 2 0 1 1 0 3 1 1 0 1 3 1 0 0 3 0 0 1 1 0 0 0 0 1 3 0 1 1 0 1 1 1 0 1 0 1 1 2 1 1 0 1 2 0 0 3 1 1 1 0 3 1 0 2 0 2 2 0 0 1 1 1 1 1 1 1 1 3 2 1 0 0 0 1 1 1 1 3 2 3 3 1 3 1 1 2 0 0 0 1 1 1 2 0 1 3 0 0 1 1 1 0 1 1 3 0 1 1 0 1 1 0 0 2 0 1 1 3 1 1 2 1 0 0 0 1 3 0 1 3 1 1 0 1 1 0 1 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 2 0 0 1 0 1 1 0 0 0 2 1 0 3 1 1 0 1 2 2 1 1 0 1 1 1 0 0 1 1 0 0 0 1 0 2 2 0 3 1 0 1 0 0 1 0 2 1 1 1 0 1 1 0 2 3 3 1 1 0 2 1 0 1 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 3 2 2 1 0 0 3 0 1 0 1 1 0 0 1 2 0 1 0 1 0 0 1 1 0 0 1 2 1 1 1 1 0 1 1 1 0 0 0 0 2 3 2 1 0 1 0 1 0 1 1 1 0 0 3 1 1 0 0 0 2 1 1 0 0 3 0 1 1 3 0 1 1 0 3 1 2 0 2 1 0 0 3 2 1 0 0 0 1 0 2 1 3 1 2 2 0 0 1 0 0 0 1 2 0 1 1 3 1 0 1 0 2 3 3 1 1 3 0 1 1 0 0 0 1 0 1 0 1 0 1 1 0 0 0 1 1 3 0 1 1 0 1 2 1 3 2\n\t [[{{node compile_loss/sparse_categorical_crossentropy/SparseSoftmaxCrossEntropyWithLogits/SparseSoftmaxCrossEntropyWithLogits}}]] [Op:__inference_one_step_on_iterator_108492976]\n"}